app/javascript/dashboard/components-next/message/bubbles/Email/Index.vue

```diff
@@ -13,41 +14,100 @@ import EmailMeta from './EmailMeta.vue';
 import { useMessageContext } from '../../provider.js';
 import { MESSAGE_TYPES } from 'next/message/constants.js';
 
+const { t } = useI18n();
+
 const { content, contentAttributes, attachments, messageType } =
   useMessageContext();
 
 const isExpandable = ref(false);
 const isExpanded = ref(false);
 const showQuotedMessage = ref(false);
+const renderOriginal = ref(false);
 const contentContainer = useTemplateRef('contentContainer');
 
 onMounted(() => {
   isExpandable.value = contentContainer.value?.scrollHeight > 400;
 });
 
-const isOutgoing = computed(() => {
-  return messageType.value === MESSAGE_TYPES.OUTGOING;
-});
+const isOutgoing = computed(() => messageType.value === MESSAGE_TYPES.OUTGOING);
 const isIncoming = computed(() => !isOutgoing.value);
 
-const textToShow = computed(() => {
+const hasTranslations = computed(() => {
+  if (!contentAttributes.value) return false;
+  const { translations = {} } = contentAttributes.value;
+  return Object.keys(translations || {}).length > 0;
+});
+
+const translationContent = computed(() => {
+  if (!hasTranslations.value) return null;
+  const translations = contentAttributes.value.translations;
+  return translations[Object.keys(translations)[0]];
+});
+
+const originalEmailText = computed(() => {
```

Comment:
Can we create a helper or composable so that we can use it text/Index.vue as well. The code seems repeated.